notebooks.go

```diff
@@ -310,57 +315,91 @@ func (s *server) GetNotebooks(w http.ResponseWriter, r *http.Request) {
 	}
 
 	for _, notebook := range notebooks {
-		// Load events
-		allevents, err := s.listers.events.Events(notebook.Namespace).List(labels.Everything())
-		if err != nil {
-			log.Printf("failed to load events for %s/%s: %v", notebook.Namespace, notebook.Name, err)
-		}
+		resp.Notebooks = append(resp.Notebooks, getNotebookData(notebook, s))
+	}
 
-		// Filter past events
-		events := make([]*corev1.Event, 0)
-		for _, event := range allevents {
-			if event.InvolvedObject.Kind != "Notebook" || event.InvolvedObject.Name != notebook.Name || event.CreationTimestamp.Before(&notebook.CreationTimestamp) {
-				continue
-			}
+	s.respond(w, r, resp)
+}
 
-			events = append(events, event)
-		}
-		sort.Sort(eventsByTimestamp(events))
+func (s *server) GetDefaultNotebook(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	namespace := vars["namespace"]
+
+	log.Printf("loading notebooks for %q", namespace)
 
-		imageparts := strings.SplitAfter(notebook.Spec.Template.Spec.Containers[0].Image, "/")
+	notebooks, err := s.listers.notebooks.Notebooks(namespace).List(labels.Everything())
+	if err != nil {
+		s.error(w, r, err)
+		return
+	}
 
-		// Process current status + reason
-		status := processStatus(notebook, events)
+	resp := &notebookapiresponse{
+		APIResponseBase: APIResponseBase{
+			Success: false,
+			Status:  http.StatusOK,
+		},
+	}
 
-		volumes := []string{}
-		for _, vol := range notebook.Spec.Template.Spec.Volumes {
-			volumes = append(volumes, vol.Name)
+	for _, notebook := range notebooks {
+		if val, ok := notebook.Labels["notebook.statcan.gc.ca/default-notebook"]; ok {
+			if val == "true" {
+				resp.Notebook = getNotebookData(notebook, s)
+				resp.APIResponseBase.Success = true
+				break
+			}
 		}
+	}
+	s.respond(w, r, resp)
+}
+
+func getNotebookData(notebook *kubeflowv1.Notebook, s *server) notebookresponse {
```

Comment:
can you declare the `s *server` like all the other functions in this file? thanks!